src/csf/sigma_vector.irp.f

```diff
@@ -52,10 +52,15 @@ end function logabsgamma
           ncfgpersomo = cfg_seniority_index(i+2)
       else
         k = 0
-        do while(cfg_seniority_index(i+2+k) < ncfgpersomo)
-          k = k + 2
-          ncfgpersomo = cfg_seniority_index(i+2+k)
-        enddo
+        if ((i+2+k) < size(cfg_seniority_index,1)) then
```

Comment:
This should work !
However, this is not the best way to do things.
Let me know if you have a better way

Reply:
Maybe we can replace the do while loop by a do loop, in order to loop between min(size(cfg_seniority_index,1) - 1, i+2) and size(cfg_seniority_index,1)-1. After we just need a condition inside the do loop to udpate ncfgpersomo or ncfgprev.
Like that:
do l = min(size(cfg_seniority_index,1) - 1, i+2), size(cfg_seniority_index,1)-1, 2
    if (cfg_seniority_index(l) >= ncfgprev) then
       ncfgprev = cfg_seniority_index(l)
    endif
enddo

It seems to work for me